framework/src/controller/channels/child_process_channel.js

```diff
@@ -142,44 +150,59 @@ class ChildProcessChannel extends BaseChannel {
 	 * @private
 	 */
 	async _resolveWhenAllSocketsBound() {
-		return Promise.all([
-			new Promise(resolve => {
-				/*
-				Here, the reason of calling .sock.once instead of pubSocket.once
-				is that pubSocket interface by Axon doesn't expose the once method.
-				However the actual socket does, by inheriting it from EventEmitter
-				prototype
-				 */
-				this.pubSocket.sock.once('connect', () => {
-					resolve();
-				});
-			}),
-			new Promise(resolve => {
-				this.subSocket.sock.once('connect', () => {
-					resolve();
-				});
-			}),
-			new Promise(resolve => {
-				this.rpcSocket.once('bind', () => {
-					resolve();
-				});
-			}),
-			new Promise((resolve, reject) => {
-				this.busRpcSocket.once('connect', () => {
-					this.busRpcClient.call(
-						'registerChannel',
-						this.moduleAlias,
-						this.eventsList.map(event => event.name),
-						this.actionsList.map(action => action.name),
-						{ type: 'ipcSocket', rpcSocketPath: this.rpcSocketPath },
-						(err, result) => {
-							if (err) reject(err);
-							resolve(result);
-						}
-					);
-				});
-			}),
-		]);
+		const promises = [];
+
+		if (this.pubSocket) {
+			promises.push(
+				new Promise(resolve => {
```

Comment:
for the sake of readability would be nice to have a function like below to promisify once events: 
```js
const promisifyOnce = (emitter, event, reject = false) => new Promise((resolve, reject) => {
  emitter.once(event, (...data) => {
    if (reject) {
      return reject(...data);
    }
    return resolve(...data);
  });
});

// Usage:

const subSocketConnectPromise = promisifyOnce(this.subSocket.sock, 'connect');
const subSocketConnectErrPromise = promisifyOnce(this.subSocket.sock, 'error', true);
```

Reply:
This code was part of another PR and I prefer keeping it as it is in this PR.